pkg/cluster/deploybaseresources_additional.go

```diff
@@ -20,7 +20,23 @@ import (
 )
 
 func (m *manager) denyAssignment() *arm.Resource {
-	return &arm.Resource{
+	excludePrincipals := []mgmtauthorization.Principal{}
+	if m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile != nil {
+		excludePrincipals = append(excludePrincipals, mgmtauthorization.Principal{
+			ID:   &m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.SPObjectID,
+			Type: to.StringPtr("ServicePrincipal"),
+		})
+	}
+	if m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile != nil {
```

Comment:
Can we keep the above conditions as `if-else`, checking for MIWI in the similar as we do:-
`if m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile != nil && m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile == nil` and keeping the `&m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.SPObjectID` in the else?

I suggest this as we had no nil check for ServicePrincipalProfile before and IMO we should consider non MIWI cluster using above check as a Service Principal cluster. It would also fail the cluster installation here that way if ServicePrincipalProfile is really nil.

Reply:
That makes sense. I've amended the PR with this